lib/maplibre-compose/src/commonMain/kotlin/org/maplibre/compose/location/LocationPuck.kt

```diff
@@ -49,59 +49,62 @@ import org.maplibre.compose.util.ClickResult
- *   be considered old locations
- * @param accuracyThreshold a circle showing the accuracy range will be drawn, when
- *   [Location.accuracy] is larger than this value. Use [Float.POSITIVE_INFINITY] to never show the
- *   accuracy range.
- * @param showBearing whether to show an indicator for [Location.bearing]
- * @param showBearingAccuracy whether to show an indicator for [Location.bearingAccuracy]
- * @param onClick a [LocationClickHandler] to invoke when the main location indicator dot is clicked
- * @param onClick a [LocationClickHandler] to invoke when the main location indicator dot is
- *   long-clicked
+ * @param idPrefix The prefix used for the layers to display the location indicator.
+ * @param location A [Location] holding the location to display.
+ * @param cameraState The [CameraState] of the map, used only for [CameraState.metersPerDpAtTarget]
+ *   to correctly draw the accuracy circle. The camera state is not modified by this composable; if
+ *   you want the camera to track the current location, use [LocationTrackingEffect].
+ * @param bearing The bearing of the location puck, which determines the rotation of the bearing
+ *   indicator. Defaults to `location.course`, which is the direction of travel.
+ * @param oldLocationThreshold Locations with a [timestamp][Location.timestamp] older than this will
+ *   be considered old and will be styled differently.
+ * @param accuracyThreshold A circle showing the accuracy range will be drawn when
+ *   [PositionMeasurement.inaccuracy] is larger than this value. Use [Float.POSITIVE_INFINITY] to
+ *   never show the accuracy range.
+ * @param colors The colors to use for the location puck.
+ * @param sizes The sizes to use for the location puck.
+ * @param onClick A [LocationClickHandler] to invoke when the main location indicator dot is
+ *   clicked.
+ * @param onLongClick A [LocationClickHandler] to invoke when the main location indicator dot is
+ *   long-clicked.
  */
 @Composable
 public fun LocationPuck(
   idPrefix: String,
-  locationState: UserLocationState,
+  location: Location?,
   cameraState: CameraState,
+  bearing: BearingMeasurement? = location?.course,
   oldLocationThreshold: Duration = 30.seconds,
   accuracyThreshold: Float = 50f,
   colors: LocationPuckColors = LocationPuckColors(),
   sizes: LocationPuckSizes = LocationPuckSizes(),
-  showBearing: Boolean = true,
-  showBearingAccuracy: Boolean = true,
   onClick: LocationClickHandler? = null,
   onLongClick: LocationClickHandler? = null,
 ) {
   val bearingPainter = rememberBearingPainter(sizes, colors)
-  val bearingAccuracyPainter =
-    rememberBearingAccuracyPainter(
-      sizes = sizes,
-      colors = colors,
-      bearingAccuracy = locationState.location?.bearingAccuracy ?: 0.0,
-    )
-  val locationSource = rememberLocationSource(locationState)
+  val locationSource = rememberLocationSource(location, bearing)
+
+  val positionAccuracy = location?.position?.inaccuracy?.inMeters?.toFloat() ?: 0f
```

Comment:
```suggestion
  val positionAccuracy = location?.position?.inaccuracy?.inMeters?.toFloat() ?: Float.NEGATIVE_INFINITY
```

So we never show the accuracy layer if we don't have a value, even if someone passes a negative `accuracyThreshold`.